pkg/conversion/mapi2capi/machine.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package mapi2capi
+
+import (
+	"fmt"
+
+	mapiv1 "github.com/openshift/api/machine/v1beta1"
+	conversionutil "github.com/openshift/cluster-capi-operator/pkg/conversion/util"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+const (
+	capiNamespace            = "openshift-cluster-api"
+	workerUserDataSecretName = "worker-user-data"
+	awsTemplateAPIVersion    = "infrastructure.cluster.x-k8s.io/v1beta2"
+	awsTemplateKind          = "AWSMachineTemplate"
+)
+
+// fromMAPIMachineToCAPIMachine translates a MAPI Machine to its Core CAPI Machine correspondent.
+func fromMAPIMachineToCAPIMachine(mapiMachine *mapiv1.Machine) (*capiv1.Machine, field.ErrorList) {
+	var errs field.ErrorList
+
+	capiMachine := &capiv1.Machine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        mapiMachine.Name,
+			Namespace:   capiNamespace,
+			Labels:      mapiMachine.Labels,
+			Annotations: mapiMachine.Annotations,
+		},
```

Comment:
Do we need the OwnerReferences here? OCPCLOUD-2716?

Reply:
On line 78 but can add here